compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/Padding.cpp

```
@@ -0,0 +1,231 @@
+  if (!isConstantIntValue(padValues[batchIdx], 0) ||
+      !isConstantIntValue(padValues[mIdx], 0) ||
+      !isConstantIntValue(padValues[nIdx], 0)) {
+    if (llvm::dyn_cast_or_null<tensor::EmptyOp>(paddedAcc.getDefiningOp())) {
+      SmallVector<OpFoldResult> paddedQueryShape =
+          tensor::getMixedSizes(rewriter, loc, paddedQuery);
+      SmallVector<OpFoldResult> paddedValueShape =
+          tensor::getMixedSizes(rewriter, loc, paddedValue);
+      SmallVector<OpFoldResult> paddedOutputShape = {
+          paddedQueryShape[0], paddedQueryShape[1], paddedValueShape[2]};
+      paddedAcc = rewriter.create<tensor::EmptyOp>(
+          loc, paddedOutputShape, attnOp.getOutputType().getElementType());
+    } else {
+      paddedAcc = getPaddedValue(
+          rewriter, loc, paddedAcc,
+          {padValues[batchIdx], padValues[mIdx], padValues[nIdx]});
+    }
+  }
+
+  // Generate padded attention op.
+  auto paddedAttnOp = rewriter.create<IREE::LinalgExt::AttentionOp>(
+      loc, paddedAcc.getType(),
+      SmallVector<Value>{paddedQuery, paddedKey, paddedValue, scale},
+      paddedAcc);
+
+  ops.push_back(paddedAttnOp);
+
+  // Extract subtensor result.
+  IntegerAttr one = rewriter.getI64IntegerAttr(1);
+  SmallVector<OpFoldResult> offsets(3, zero);
+  SmallVector<OpFoldResult> strides(3, one);
+  SmallVector<OpFoldResult> sizes =
+      tensor::getMixedSizes(rewriter, loc, attnOp.getOutput());
+  Operation *extracted = rewriter.create<tensor::ExtractSliceOp>(
+      loc, paddedAttnOp->getResults()[0], offsets, sizes, strides);
+  ops.push_back(extracted);
+
+  rewriter.replaceOp(attnOp, extracted);
+
+  return DiagnosedSilenceableFailure::success();
+}
+
+void PadAttentionPass::runOnOperation() {
+  MLIRContext *context = &getContext();
+  IRRewriter rewriter(context);
+  getOperation().walk([&](AttentionOp attnOp) {
+    SmallVector<Operation *> ops;
+    (void)padAttention(attnOp, ops, rewriter, std::nullopt, padToMultipleOf);
```

Maybe catch the error and fail here instead of just ignoring errors.